### PR TITLE
Blog App: Validations and model specs

### PR DESCRIPTION
**Pull request by gitbygaurav**
### **Hi CR 👋**

HelloDogedashGIF

**In this branch I have done the following changes:-**

- [x] I have added the following validations:
- [x] For the User model:
 

> Name must not be blank.
PostsCounter must be an integer greater than or equal to zero.

- [x] For the Post model:

> Title must not be blank.
The title must not exceed 250 characters.
CommentsCounter must be an integer greater than or equal to zero.
LikesCounter must be an integer greater than or equal to zero.

- [x] I also added unit specs for all of your models' methods and validations.

Please review my changes 
Thanks🍻

### CODE REVIEW

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  user = User.create(name: 'Grabrielle', photo: 'mybaby.png', bio: 'A beautiful baby', post_counter: 0)
+  post = Post.create(user_id: author.id, title: 'Hello', text: 'This is my first post', likes_counter: 0,
+                     comments_counter: 0)
```
Comment (KDlamini) [inline spec/models/comment_spec.rb]: Some tests for comments and likes are failing due to the following reasons.

image

-  On both the `comment_spec.rb` and `like_spec.rb` `line: 5` the variable `author` does not exit therefore `user_id: author.id` should be `user_id: user.id`.

Kindly fix all the issues with foreign keys on your test to make sure that all your tests pass.

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @gauravgangwarweb 🙋‍♀️,  

Great work so far! 💯 
You have done a great job adding model validations and validation specs for the Blog App. 🎊 
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 

✔️  Giood PR title and description
✔️  Model validations are well implemented
✔️  All validations specs for each model are covered

## Required Changes ♻️
 
Please make your README more professional by adding information about:

 -  How to run the project `rails s or rails server` under the `Getting Started` section and `Usage` sub-heading.
 -  How to run the tests `bundle exec rspec ./spec` under the `Getting Started` section and `Run tests` sub-heading.
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb (hunk 1 above)
Comment (peterrobert) [reply spec/models/comment_spec.rb]: This issue is not yet resolved

File: spec/models/comment_spec.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 RSpec.describe Comment, type: :model do
   user = User.create(name: 'Grabrielle', photo: 'mybaby.png', bio: 'A beautiful baby', post_counter: 0)
-  post = Post.create(user_id: author.id, title: 'Hello', text: 'This is my first post', likes_counter: 0,
+  post = Post.create(user_id: user.id, title: 'Hello', text: 'This is my first post', likes_counter: 0,
                      comments_counter: 0)
 
   comment = Comment.new(post_id: post.id, user_id: user.id, text: 'Hi Tom!')
```
Comment (peterrobert) [inline spec/models/comment_spec.rb]: - [ ] It seems this issue was not fully resolved. There seem to be errors popping up when you run the test.  I have linked a screenshot of the errors. Kindly solve this issue. 🔽 
test

**Review by peterrobert — CHANGES_REQUESTED** [review]
Hi @gauravgangwarweb, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏


Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Kamba56 — APPROVED** [review]
Hi @gauravgangwarweb, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
## Highlights
- [x] Good job adding tests for validations!👏👏👏
- [x] Great job adding tests for custom methods!👏👏👏
- [x] Descriptive PR!👏👏👏

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**